extensions/wikia/UserLogin/UserLoginHooksHelper.class.php

```diff
@@ -116,7 +116,12 @@ public static function onSetUserEmail( User $user, $newEmail, &$result, &$info )
 	}
 
 	public static function isValidEmailAddr( $addr ) {
-		return preg_match( '/^[a-z0-9._%+-]+@(?:[a-z0-9\-]+\.)+[a-z]{2,4}$/i', $addr ) !== 0;
+		// this validation match Helios email validation
+		// [see: https://github.com/Wikia/helios/blob/ce490394ae10eb248afd2a063dc465d8eb4efb01/domain/user_validator_funs.go#L29]
+		return preg_match(
+			'/^[\w!#$%&\'*+\/=?^_`{|}~-]+(?:\.[\w!#$%&\'*+\/=?^_`{|}~-]+)*@(?:[\w](?:[\w-]*[\w])?\.)+[a-zA-Z0-9](?:[\w-]*[\w])?$/i',
+			$addr
+		) !== 0;
```

Comment:
We shouldn't replace one hand made email regex with another.  The actual email validation regex is lengthy and we're guaranteed to miss something if we try our own.  PHP has a validation function that can be used:
```
http://php.net/manual/en/filter.examples.validation.php
```
So, looks like:
```
if (filter_var($email, FILTER_VALIDATE_EMAIL)) { ... }
```
I've asked the Helios team to update their code to use the Go library rather than this lib as well.